File: 5th Semester/mobile/swift/Model/Car.swift
//
//  Car.swift
//  Rental
//
//  Created by Mircea Maierean on 05.11.2024.
//

import Foundation
import SwiftData

struct Car: Identifiable, Decodable {
    var id: Int
    var make: String
    var model: String
    var color: String
    var productionYear: Int
    var carburantType: String
    var bodyType: String
    var pricePerDay: Int
    var photo: String?  
}

File: 5th Semester/mobile/localdb/rental/lib/models/car.dart
class Car {
  final int id;
  final String make;
  final String model;
  final String color;
  final int productionYear;
  final String carburantType;
  final String bodyType;
  final int pricePerDay;
  final String? photoPath;

  Car({
    required this.id,
    required this.make,
    required this.model,
    required this.color,
    required this.productionYear,
    required this.carburantType,
    required this.bodyType,
    required this.pricePerDay,
    this.photoPath,
  });

  Map<String, dynamic> toMap() {
    return {
      'id': id,
      'make': make,
      'model': model,
      'color': color,
      'productionYear': productionYear,
      'carburantType': carburantType,
      'bodyType': bodyType,
      'pricePerDay': pricePerDay,
      'photoPath': photoPath,
    };
  }

  factory Car.fromMap(Map<String, dynamic> map) {
    return Car(
      id: map['id'],
      make: map['make'],
      model: map['model'],
      color: map['color'],
      productionYear: map['productionYear'],
      carburantType: map['carburantType'],
      bodyType: map['bodyType'],
      pricePerDay: map['pricePerDay'],
      photoPath: map['photoPath'],
    );
  }
}
